### ISSUE TITLE
Installation instructions have and produce errors

### ISSUE THREAD

**Issue by SethM55** [PacificBiosciences/blasr#87]
Hello,

I've been trying to get Blasr running for a few days now and checked the other issues (closed or open) and it appears I'm not alone with installation problems.

1) The git command you give is invalid, as pointed out in issue #85 the URL is incorrect, but the readme.md and manual.md files have not been updated.

2) If using the git pull, the next step is checking that hdf5 has been installed.  A quick locate told me they were already in the shared  usr/include and /usr/share directories.  The documentation says I can export to path or edit lasr_git_common.mk, but there is no lasr_git_common.mk file in the pulled directory.

3) Sticking with the git pull the next step should be a make or make blasr.  In either case this generates the error: "blasr/defines.mk: No such file or directory"  this file is missing from the git pull it seems.

4) If using the github zip download button lasr_git_common.mk can be found and edited.  running the make this time gives the error: "/blasr-master/libcpp/common.mk: No such file or directory"  the libcpp folder was empty, which was my fault because I used the zip download instead of git to pull the files.

5) Using the zip download on the libcpp files the error still occurs, so I need to compile libcpp before I can make blasr.  According to the libcpp readme.md this is done by running ./configure.py which mentions no errors on execution.  the next step is "make -j all" which produces a stream of the same error in multiple files:
./DNASequence.hpp:14:29: fatal error: pbbam/BamRecord.h: No such file or directory
 #include "pbbam/BamRecord.h"
                             ^
compilation terminated.
make[1]: *** [SMRTSequence.o] Error 1
make[1]: *** [FASTAReader.o] Error 1
make[1]: *** [PackedDNASequence.o] Error 1
make[1]: *** [CCSSequence.o] Error 1
make[1]: *** [FASTQReader.o] Error 1
make[1]: *** [reads/PulseFile.o] Error 1
make[1]: *** [FASTASequence.o] Error 1
make[1]: *** [utils/SMRTReadUtils.o] Error 1
make[1]: *** [FASTQSequence.o] Error 1
make[1]: *** [reads/BaseFile.o] Error 1
make[1]: Leaving directory `/genome/Apps/blasr-master/libcpp/pbdata'
make: *** [all] Error 2

pbbam/BamRecord.h is always the issue, so it seems that this make command is not actually skipping the pbbam as the readme.md says.

6) I tried copying the defines.mk file from the zip download into the git pull and got the error:
MappingParameters.h:9:30: fatal error: reads/ReadType.hpp: No such file or directory
 #include "reads/ReadType.hpp"

At this point neither approach (git pull or zip download) can actually move forward and be installed.  I'd like to get the git pull to work, but I don't know how to make it find/build the correct defines.mk.  Suggestions?

**Comment by pb-cdunn** [comment]
1) Fixed README in b7abc8be353e4d74b4e712d5aa7620906c8e3f5f. I cannot find `manual.md`. Which commit are you using?
2)
>  The documentation says I can export to path or edit blasr_git_common.mk, but there is no blasr_git_common.mk file in the pulled directory.

That's wrong. Is this in the wiki? If so, we need to correct that. Can you point us to a URL?

3) `blasr/defines.mk` is generated by `blasr/configure.py`. See [travis.sh](../blob/master/travis.sh) for an example of how to use `configure.py`.
4) Which github zip? What URL?
5) Instead of running `libcpp/configure.py`, run `./configure.py` in the `blasr/` directory. Then, run `make init-submodule` to build libcpp. See [travis.sh](../blob/master/travis.sh).
6) `defines.mk` is generated by convention. It is a different file in blasr than in libcp, so copying it will not work.

**Comment by SethM55** [comment]
Thanks for your help!

1) I used <code>git clone https://github.com/PacificBiosciences/blasr.git --recursive</code> and it had <code>README.md</code> which opened with the lines:
<code>Blasr installation and maual:
<p>
    more Manual.md</code>
So I opened <code>MANUAL.md</code> found in the root blasr directory.

2) This line is in the <code>MANUAL.md</code> file

3) There was no mention in <code>MANUAL.md</code> to run <code>configure.py</code> in the blasr root directory, so I failed to do this, probably why this isn't working.

4) On the website <code>https://github.com/PacificBiosciences/blasr</code> there is a <code>Download ZIP</code> button on the right column provided by github that lets you download any git (or at least all of them that I've seen) as a zip.

5) I started with a clean git pull, to be safe, and ran the commands in Travis.sh, but skipped the hdf5 parts (<code>./configure.py --shared --sub --no-pbbam</code>) because a <code>locate</code> tells me the HDF5 files are in <code>/usr/include/</code> and <code>/usr/lib/</code> but <code>make init-submodule</code> failed because it was looking for HDF5 files in <code>/</code>.  Does the make not check the standard shared and include folders?

So I ran <code>./configure.py --shared --sub --no-pbbam HDF5_INCLUDE=/usr/include/ HDF5_LIB=/usr/lib/</code> then <code>make -j4 init-submodule</code> and got new set of the error
<code>error: ‘H5Location’ in namespace ‘H5’ does not name a type<p>
     H5::H5Location* GetObject();</code>
I'm guessing this means there's something not right with my HDF5 libraries?

6) I figured it would be, but it was a desperation move.

**Comment by pb-cdunn** [comment]
1) Oh, case-sensitive filesystem. I hadn't noticed MANUAL.md. Fixing...

4) We will have a new release soon. Until then, old releases relied on super-complicated makefiles, so they may or may not work. It's better to use the tip of the master branch. I've updated the release-notes.

5) You might have an ancient HDF5 installation. Install the latest, and try again.

A good way to debug integration problems is to alter `defines.mk` directly. Then, you can run `configure.py` in a way that creates the `defines.mk` that works for you. If you need to alter `configure.py`, just submit a pull-request. We cannot anticipate every integration problem, but we are willing to work with you.

**Comment by pb-cdunn** [comment]
Updated READMEs and wiki. If you need help with HDF5 problems, please open a new issue.

**Comment by SethM55** [comment]
Huzzah for corrections!

Apt-get said I was using the latest version.  Just to be sure, I downloaded and compiled from source via the hdf group website (https://www.hdfgroup.org/HDF5/) their README mentioned that the default setup did NOT include C++ interfaces.  I added the required <code>--enable-cxx</code> to the <code>configure</code> command and it worked.  Going back to the travis.sh commands compiled without errors and I now have a working version of blasr.

Thanks again for your help!
